pennylane_sf/remote.py

```diff
@@ -0,0 +1,152 @@
+from strawberryfields.utils.post_processing import (all_fock_probs_pnr,
+                                                    samples_expectation,
+                                                    samples_variance)
+
+from .simulator import StrawberryFieldsSimulator
+
+
+class StrawberryFieldsRemote(StrawberryFieldsSimulator):
+    r"""StrawberryFields remote device for PennyLane.
+
+    A valid Strawberry Fields API token is needed for access. This token can be
+    passed when creating the device. The default configuration options of
+    Strawberry Fields are used to store the authentication token in a
+    configuration file.
+
+    Args:
+        wires (int): the number of modes to initialize the device in
+        shots (int): number of circuit evaluations/random samples used to
+            estimate expectation values of observables
+        backend (str): name of the remote backend to be used
+        hbar (float): the convention chosen in the canonical commutation
+            relation :math:`[x, p] = i \hbar`
+        sf_token (str): the SF API token used for remote access
+    """
+    name = "Strawberry Fields Hardware PennyLane plugin"
+    short_name = "strawberryfields.ai"
+
+    _operation_map = {
+        'CatState': Catstate,
+        'CoherentState': Coherent,
+        'FockDensityMatrix': DensityMatrix,
+        'DisplacedSqueezedState': DisplacedSqueezed,
+        'FockState': Fock,
+        'FockStateVector': Ket,
+        'SqueezedState': Squeezed,
+        'ThermalState': Thermal,
+        'GaussianState': Gaussian,
+        'Beamsplitter': BSgate,
+        'CrossKerr': CKgate,
+        'ControlledAddition': CXgate,
+        'ControlledPhase': CZgate,
+        'Displacement': Dgate,
+        'Kerr': Kgate,
+        'QuadraticPhase': Pgate,
+        'Rotation': Rgate,
+        'TwoModeSqueezing': S2gate,
+        'Squeezing': Sgate,
+        'CubicPhase': Vgate,
+        'Interferometer': Interferometer
+    }
+
+    _observable_map = {
+        "Identity": None,
+        "NumberOperator": None,
+        "TensorN": None,
+    }
+
+    def __init__(self, wires, *, backend, shots=1000, hbar=2, sf_token=None):
+        super().__init__(wires, analytic=False, shots=shots, hbar=hbar)
+        self.backend = backend
+
+        if sf_token is not None:
+            sf.store_account(sf_token)
+
+    def pre_measure(self):
+        self.eng = sf.RemoteEngine(self.backend)
+
+        MeasureFock() | self.q  # pylint: disable=pointless-statement, expression-not-assigned
+
+        # RemoteEngine.run includes compilation that checks the validity of the
+        # defined Program
+        results = self.eng.run(self.prog, shots=self.shots)
+        self.samples = results.samples
+
+    def sample(self, observable, wires, par): # pylint: disable=unused-argument, missing-function-docstring
+        wires = np.array(wires)
+        selected_samples = self.samples[:, wires]
+        return np.prod(selected_samples, axis=1)
```

Comment:
This only applies for the `TensorN` and `NumberOperator` observables. I wonder how we would compute `<I>` for the `Identity` observable 🤔

Reply:
For the time being, if `observable == "Identity"`, lets just return 1.